src/deluge/model/clip/instrument_clip.cpp

```diff
@@ -2729,99 +2701,127 @@ Error InstrumentClip::readFromFile(Deserializer& reader, Song* song) {
 			while (*(tagName = reader.readNextTagOrAttributeName())) {
 
 				if (!strcmp(tagName, "rate")) {
-					arpeggiatorRate = reader.readTagOrAttributeValueInt();
+					arpSettings.rate = reader.readTagOrAttributeValueInt();
```

Comment:
Reading can be shared `ArpeggiatorSettings::readTagFromFile(Deserializer&  reader, char const * tagName)`.`